pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/design/xml/ExceptionAsFlowControl.xml

```diff
@@ -95,4 +98,70 @@ class SubE extends TopE { }
 
         ]]></code>
     </test-code>
+    <test-code>
+        <description>#4434 simple propagation</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+            import java.util.*;
+            public class Foo {
+
+                public Map<K, V> getAll(Iterable<? extends K> keys) throws ExecutionException {
+                    try {
+                        Map<K, V> result = cache.getAll(keys);
+                        if (something()) {
+                            throw new InvalidCacheLoadException("null key or value");
+                        }
+                        return result;
+                    } catch (NullPointerException | InvalidCacheLoadException e) {
+                        throw e; // <-- Violation
+                    } catch (CacheLoaderException e) {
+                        throw new ExecutionException(e.getCause());
+                    } catch (Exception e) {
+                        throw new UncheckedExecutionException(e);
+                    }
+                }
+            }
+
+            class ExecutionException extends Exception {
+                public ExecutionException(Throwable o) { super(o);}
+            }
+            class UncheckedExecutionException extends RuntimeException {
+                public UncheckedExecutionException(Throwable o) { super(o);}
+            }
+            class InvalidCacheLoadException extends RuntimeException {
+                public InvalidCacheLoadException(String msg) { super(msg);}
+            }
+
+            ]]></code>
+    </test-code>
+    <test-code>
+        <description>#4434 simple propagation, 2nd case</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+            import java.util.*;
+            public class Foo {
+
+                public Map<K, V> getAll(Iterable<? extends K> keys) {
+                    try {
+                        Map<K, V> result = cache.getAll(keys);
+                        if (something()) {
+                            throw new InvalidCacheLoadException("null key or value");
+                        }
+                        return result;
+
+                    } catch (NullPointerException | InvalidCacheLoadException e) {
+                        // Note that technically this is still an code smell as
+                        // the exception is rethrown directly, but this should be
+                        // the purpose of another rule.
+                        throw e;
+                    }
```

Comment:
is it ok to whitelist any multicatch blindly?

the original issue was limited on "without the catch, the exception would be processed differently" (with IMHO should be handled by an `if` rather than a new `catch` block), but on this case, the only thing stopping this from being reported is the use of multi-catch… and having:

```java
catch (NullPointerException e) {
    throw e;
} catch (InvalidCacheLoadException e) {
    throw e;
}
```

would be reported twice… I find it very inconsistent to be honest